src/arch/cuda/hybridAPI/hapi_impl.cpp

```diff
@@ -515,10 +512,10 @@ static void* hostToDeviceCallback(void* arg) {
   NVTXTracer nvtx_range("hostToDeviceCallback", NVTXColor::Asbestos);
 #endif
   hapiWorkRequest* wr = *((hapiWorkRequest**)((char*)arg + CmiMsgHeaderSizeBytes + sizeof(int)));
-  CUDACallbackManager(wr->host_to_device_cb);
+  hapiInvokeCallback(wr->host_to_device_cb);
 
   // inform QD that the host-to-device transfer is complete
-  QdProcess(1);
+  hapiQdProcess(1);
```

Comment:
These function calls might crash if the function pointers haven't been initialized in `_initCharm`. Maybe add a check in `initHybridAPI` that these pointers are not `NULL` and `CkAbort` the application if they are.